### PR TITLE
[stdlib] Constrain Lazy.Elements.Element == Element

### PR DESCRIPTION
**Pull request by airspeedswift**

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

File: stdlib/public/core/Flatten.swift.gyb [hunk 1]
```diff
@@ -418,11 +415,11 @@ extension LazyCollectionProtocol
...
-  ${constraints % {'Base': 'Elements.'}},
-  Iterator.Element == Elements.Iterator.Element {
-  /// A concatenation of the elements of `self`.
-  public func joined() -> LazyCollection<${Collection}<Elements>> {
-    return ${Collection}(elements).lazy
+  ${constraints % {'Base': 'Elements.'}}
+ {
+    /// A concatenation of the elements of `self`.
+    public func joined() -> LazyCollection<${Collection}<Elements>> {
+      return ${Collection}(elements).lazy
```
Comment (xwu) [inline stdlib/public/core/Flatten.swift.gyb]: Drive-by nit: The indentation here got out of whack.

Comment (airspeedswift) [inline stdlib/public/core/Flatten.swift.gyb]: oop, thanks

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge
